kong/client_test.go

```diff
@@ -205,3 +207,123 @@ func TestBaseRootURL(t *testing.T) {
 		require.Equal(t, client.BaseRootURL(), "https://customkong2.com")
 	})
 }
+
+func TestReloadDeclarativeRawConfig(t *testing.T) {
+	RunWhenDBMode(t, "off")
+
+	tests := []struct {
+		name    string
+		config  Configuration
+		wantErr bool
+	}{
+		{
+			name: "basic config works",
+			config: Configuration{
+				"_format_version": "1.1",
+				"services": []Configuration{
+					{
+						"host":     "mockbin.com",
+						"port":     443,
+						"protocol": "https",
+						"routes": []Configuration{
+							{"paths": []string{"/"}},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "missing _format_version fails",
+			config: Configuration{
+				"services": []Configuration{
+					{
+						"host":     "mockbin.com",
+						"port":     443,
+						"protocol": "https",
+						"routes": []Configuration{
+							{"paths": []string{"/"}},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid config fails",
+			config: Configuration{
+				"dummy_key": []Configuration{
+					{
+						"host":     "mockbin.com",
+						"port":     443,
+						"protocol": "https",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	// POST /config actually interprets query string fields as additional fields within the config unless they're
+	// explicitly stripped by the API code (no idea why this behavior exists). Without stripping "flatten_errors" in
+	// https://github.com/Kong/kong/blob/master/kong/api/routes/config.lua#L71 Kong will actually reject configuration
+	// because it thinks "flatten_errors" is an actual (and invalid) field inside the config.
+	//
+	// This is the one test where we want version-dependent behavior, but only for one value within the test. We test
+	// config updates on all DB-less capable versions, but only set flattenErrors=true on 3.2+. To handle that, this
+	// snippet is borrowed from RunWhenKong, to allow toggling that behavior only on or off depending on the version.
+	var flattenErrors bool
+	client, err := NewTestClient(nil, nil)
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
Why no `require.NoError(t, err)`? Much less boilerplate

Ditto elsewhere

Reply:
Literally just copy-pasta from the source. No idea why the helper's written that way (we do generally want to avoid `require`s in helpers, but that's a more broad advice against avoiding any interaction with the test struct in helpers, and we were already doing that with the `t.Error(err)`s.